### PR TITLE
Add details to the summary for show more error informations

### PR DESCRIPTION
**Pull request by Avin0323**
### PR types
Others

### PR changes
Others

### Describe
Collect error informations via `stdout` and used as argument of `summary_check_problems` to show.

There is a test log in [PR-CI-CPU-Py2](https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/1692831/job/2420361).
image

### CODE REVIEW

**Comment by luotao1** [comment]
```
2020-09-08 14:23:39 ========================================
2020-09-08 14:23:39 summary problems:
2020-09-08 14:23:39 ========================================
2020-09-08 14:23:39 - Check code style failed! Error infomations follow:
2020-09-08 14:23:39 code format error
2020-09-08 14:23:39 diff --git a/python/paddle/fluid/tests/unittests/test_linspace.py b/python/paddle/fluid/tests/unittests/test_linspace.py
2020-09-08 14:23:39 index c9c099d..03cb84e 100644
2020-09-08 14:23:39 --- a/python/paddle/fluid/tests/unittests/test_linspace.py
2020-09-08 14:23:39 +++ b/python/paddle/fluid/tests/unittests/test_linspace.py
2020-09-08 14:23:39 @@ -156,7 +156,7 @@ class TestLinspaceOpError(unittest.TestCase):
2020-09-08 14:23:39              def test_start_dtype():
2020-09-08 14:23:39                  start = fluid.data(shape=[1], dtype="float64", name="start")
2020-09-08 14:23:39                  fluid.layers.linspace(start, 10, 1, dtype="float32")
2020-09-08 14:23:39 -            
2020-09-08 14:23:39 +
2020-09-08 14:23:39              self.assertRaises(ValueError, test_start_dtype)
2020-09-08 14:23:39  
2020-09-08 14:23:39              def test_end_dtype():
2020-09-08 14:23:39 - Check example code failed! Error informations follow:
2020-09-08 14:23:39 API check -- Example Code
2020-09-08 14:23:39 ('sample_test running under python', '2.7.15')
2020-09-08 14:23:39 API_PR is diff from API_DEV: ['../python/paddle/fluid/nets.py']
2020-09-08 14:23:39 ('def', ' name:', 'simple_img_conv_pool')
2020-09-08 14:23:39 -----------------------
2020-09-08 14:23:39 Error: No sample code!
2020-09-08 14:23:39 
2020-09-08 14:23:39 ----------------End of the Check--------------------
2020-09-08 14:23:39 Mistakes found in sample codes.
2020-09-08 14:23:39 Please check sample codes.
```
从日志看：
1. 这里有两个错误，第一个是代码格式，第二个是示例错误。但两行提醒不够明显。应该首先说有2个错误，是xxx和xxx，然后再分别空行或者****打出xxx对应的具体错误
```
- Check code style failed! Error infomations follow
```
2. 代码格式错误，目前日志里报的是告诉用户怎么改。那就需要显示的告诉用户，不然大家看不懂下面这段说的是什么，以为是错误代码

**Comment by Avin0323** [comment]
> ```
> 2020-09-08 14:23:39 ========================================
> 2020-09-08 14:23:39 summary problems:
> 2020-09-08 14:23:39 ========================================
> 2020-09-08 14:23:39 - Check code style failed! Error infomations follow:
> 2020-09-08 14:23:39 code format error
> 2020-09-08 14:23:39 diff --git a/python/paddle/fluid/tests/unittests/test_linspace.py b/python/paddle/fluid/tests/unittests/test_linspace.py
> 2020-09-08 14:23:39 index c9c099d..03cb84e 100644
> 2020-09-08 14:23:39 --- a/python/paddle/fluid/tests/unittests/test_linspace.py
> 2020-09-08 14:23:39 +++ b/python/paddle/fluid/tests/unittests/test_linspace.py
> 2020-09-08 14:23:39 @@ -156,7 +156,7 @@ class TestLinspaceOpError(unittest.TestCase):
> 2020-09-08 14:23:39              def test_start_dtype():
> 2020-09-08 14:23:39                  start = fluid.data(shape=[1], dtype="float64", name="start")
> 2020-09-08 14:23:39                  fluid.layers.linspace(start, 10, 1, dtype="float32")
> 2020-09-08 14:23:39 -            
> 2020-09-08 14:23:39 +
> 2020-09-08 14:23:39              self.assertRaises(ValueError, test_start_dtype)
> 2020-09-08 14:23:39  
> 2020-09-08 14:23:39              def test_end_dtype():
> 2020-09-08 14:23:39 - Check example code failed! Error informations follow:
> 2020-09-08 14:23:39 API check -- Example Code
> 2020-09-08 14:23:39 ('sample_test running under python', '2.7.15')
> 2020-09-08 14:23:39 API_PR is diff from API_DEV: ['../python/paddle/fluid/nets.py']
> 2020-09-08 14:23:39 ('def', ' name:', 'simple_img_conv_pool')
> 2020-09-08 14:23:39 -----------------------
> 2020-09-08 14:23:39 Error: No sample code!
> 2020-09-08 14:23:39 
> 2020-09-08 14:23:39 ----------------End of the Check--------------------
> 2020-09-08 14:23:39 Mistakes found in sample codes.
> 2020-09-08 14:23:39 Please check sample codes.
> ```
> 
> 从日志看：
> 
> 1. 这里有两个错误，第一个是代码格式，第二个是示例错误。但两行提醒不够明显。应该首先说有2个错误，是xxx和xxx，然后再分别空行或者****打出xxx对应的具体错误
> 
> ```
> - Check code style failed! Error infomations follow
> ```
> 
> 1. 代码格式错误，目前日志里报的是告诉用户怎么改。那就需要显示的告诉用户，不然大家看不懂下面这段说的是什么，以为是错误代码

done @luotao1

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -1486,15 +1496,16 @@ function main() {
...
         cmake_gen_and_build ${PYTHON_ABI:-""} ${parallel_number}
         check_sequence_op_unittest
         generate_api_spec ${PYTHON_ABI:-""} "PR"
-        $(example >&2)
+        set +e
+        example_info=$(example)
         example_code=$?
-        summary_check_problems $check_style_code $example_code
+        summary_check_problems $check_style_code $example_code "$check_style_info" "$example_info"
         assert_api_spec_approvals
```
Comment (lelelelelez) [inline paddle/scripts/paddle_build.sh:1509]: summary_check_problems 汇总之后，假如有错，直接退出了。
api的approval是放到下个pr做吗？

Comment (luotao1) [reply paddle/scripts/paddle_build.sh:1509]: 可以放到下一个PR做 @Avin0323

**Review by luotao1 — APPROVED** [review]

**Pull request merged**